components/Header/Header.tsx

```diff
@@ -0,0 +1,56 @@
+import ToggleTheme from "./ToggleTheme";
+import { Avatar } from "@mui/material";
+import { auth } from "../../firebase";
+import Logo from "../Logo"
+import Link from 'next/link'
+import SearchBar from './SearchBar';
+import SettingsButton from './SettingsButton'
+import AppsButton from './AppsButton';
+import NavLinks from './NavLinks';
+
+const Header = () => {
+  return (
+    <div>
+        {/* Top: Toolbar */}
+        <div 
+        className='grid grid-cols-2 md:grid-cols-3 gap-y-2 grid-flow-row-dense w-full 
+        py-5 px-3 justify-items-stretch items-center bg-white dark:bg-neutralDark-500'>
+            {/* Left: Logo */}
+            <div 
+            className="flex items-center justify-self-start cursor-pointer md:mr-auto
+            text-black dark:text-neutralDark-50">
+                <Link href="/" passHref>
+                    <a><Logo fill="currentColor"/></a>
+                </Link>
+            </div>
+        
+            {/* Center: Search */}
+            <div 
+            className="flex items-center justify-center col-span-2 md:col-span-1 md:justify-items-start">
+                <SearchBar placeholder="What's your question?" />
+            </div>
+        
+            {/* Right: User */}
+            <div 
+            className="flex space-x-2 items-center justify-self-end whitespace-nowrap px-1 md:ml-auto
+            text-neutral-700 dark:text-neutralDark-150">
+                <SettingsButton />
+                <AppsButton />
```

Comment:
We probably want this only on md and above. Or we can ask designers to design a mobile version.

Reply:
Are you suggesting we completely hide the user tool bar on mobile? Users will need a way to access this functionality on mobile (obvi. a dedicated mobile app is ideal, but we need browser-based support on mobile too).

What's your issue with the current layout? Too crowded? I think the three row navbar is a decent trade off on most mobile devices. Only on the smallest Apple and Samsung screen does it take up more than ~1/5 height of the screen.

One solution would be to hide rows 2/3 on small screens and have a drop toggle to expose the search bar and nav links.

Reply:
Gets crowded. Look at google layout on small screen. On mobile, you can hide the icons under the avatar and reduce the font size for the 3 tabs. Also, we won't have search page.